src/platform/tests/MpiServerTests.cpp

```diff
@@ -0,0 +1,234 @@
+        EXPECT_EQ(HTTP_BAD_REQUEST, HandleMpiCall(MPI_SET_DESIRED_URI, "{\"Payload\": {}}", &response, &responseSize, g_mpiCalls));
+        EXPECT_EQ(HTTP_BAD_REQUEST, HandleMpiCall(MPI_SET_DESIRED_URI, "{\"ClientSession\": \"\"}", &response, &responseSize, g_mpiCalls));
+
+        EXPECT_EQ(HTTP_BAD_REQUEST, HandleMpiCall(MPI_GET_REPORTED_URI, "{\"ClientSession\": 123}", &response, &responseSize, g_mpiCalls));
+        EXPECT_EQ(HTTP_BAD_REQUEST, HandleMpiCall(MPI_GET_REPORTED_URI, "{}", &response, &responseSize, g_mpiCalls));
+    }
+
+    TEST_F(MpiServerTests, MpiCloseRequest)
+    {
+        char* response = nullptr;
+        int responseSize = 0;
+
+        EXPECT_EQ(HTTP_OK, HandleMpiCall(MPI_CLOSE_URI, "{\"ClientSession\": \"Valid_Client\"}", &response, &responseSize, g_mpiCalls));
+    }
+
+    TEST_F(MpiServerTests, MpiSetRequest)
+    {
+        char* response = nullptr;
+        int responseSize = 0;
+
+        EXPECT_EQ(HTTP_INTERNAL_SERVER_ERROR, HandleMpiCall(MPI_SET_URI, "{\"ClientSession\": \"Valid_Client\", \"ComponentName\": \"Error_Component\", \"ObjectName\": \"Error_Object\", \"Payload\": {}}", &response, &responseSize, g_mpiCalls));
+        EXPECT_EQ(nullptr, response);
+        EXPECT_EQ(0, responseSize);
+
+        EXPECT_EQ(HTTP_OK, HandleMpiCall(MPI_SET_URI, "{\"ClientSession\": \"Valid_Client\", \"ComponentName\": \"\", \"ObjectName\": \"\", \"Payload\": {}}", &response, &responseSize, g_mpiCalls));
+        EXPECT_EQ(nullptr, response);
+        EXPECT_EQ(0, responseSize);
+    }
+
+    TEST_F(MpiServerTests, MpiGetRequest)
+    {
+        char* response = nullptr;
+        int responseSize = 0;
+
+        EXPECT_EQ(HTTP_INTERNAL_SERVER_ERROR, HandleMpiCall(MPI_GET_URI, "{\"ClientSession\": \"Valid_Client\", \"ComponentName\": \"Error_Component\", \"ObjectName\": \"Error_Object\"}", &response, &responseSize, g_mpiCalls));
+        EXPECT_EQ(nullptr, response);
+        EXPECT_EQ(0, responseSize);
+
+        EXPECT_EQ(HTTP_OK, HandleMpiCall(MPI_GET_URI, "{\"ClientSession\": \"Valid_Client\", \"ComponentName\": \"\", \"ObjectName\": \"\"}", &response, &responseSize, g_mpiCalls));
+        EXPECT_STREQ(g_mockPayload, response);
+        EXPECT_EQ(strlen(g_mockPayload), responseSize);
+    }
+
+    TEST_F(MpiServerTests, MpiSetDesiredRequest)
+    {
+        char* response = nullptr;
+        int responseSize = 0;
+
+        EXPECT_EQ(HTTP_INTERNAL_SERVER_ERROR, HandleMpiCall(MPI_SET_DESIRED_URI, "{\"ClientSession\": \"Valid_Client\", \"Payload\": {}}", &response, &responseSize, g_mpiCalls));
+        EXPECT_EQ(nullptr, response);
+        EXPECT_EQ(0, responseSize);
+
+        EXPECT_EQ(HTTP_OK, HandleMpiCall(MPI_SET_DESIRED_URI, "{\"ClientSession\": \"Valid_Client\", \"Payload\": \"MockPayload\"}", &response, &responseSize, g_mpiCalls));
+        EXPECT_EQ(nullptr, response);
+        EXPECT_EQ(0, responseSize);
+    }
+
+    TEST_F(MpiServerTests, MpiGetReportedRequest)
+    {
+        char* response = nullptr;
```

Comment:
>char* response = nullptr; [](http://example.com/codeflow?start=8&length=25)

Where is this freed? Same for all other similar instances

Reply:
Added `FREE_MEMORY` to each of the tests where memory is allocated.